### PR TITLE
Add Alternate key E2E

### PR DESCRIPTION
**Pull request by xuzhg**
Modify the Product codes to make the alternate keys routing works.
Add many alternate key e2e test cases.

### CODE REVIEW

File: OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+            Assert.Equal(expect, responseContent);
+        }
+
+        [Fact]
+        public async Task QueryEntityWithSingleAlternateKeysWorks()
+        {
+            // query with alternate keys
+            const string expect = @"{
+  ""@odata.context"":""http://sam-xu-odata:9001/odata/$metadata#Edm.String"",""value"":""special-SSN""
```
Comment (abkmr) [inline OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs]: Need to remove the odata.context which will cause it to fail on another machine during E2e test runs.

File: OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            Console.WriteLine(requestUri);
+            Console.WriteLine(responseContent);
+        }
+         * */
+
+        [Fact]
+        public async Task CanUpdateEntityWithSingleAlternateKeys()
+        {
+            const string expect = @"{
+  ""@odata.context"":""http://sam-xu-odata:9001/odata/$metadata#Customers/$entity"",""ID"":6,""Name"":""Updated Customer Name"",""SSN"":""SSN-6-T-006""
```
Comment (abkmr) [inline OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs]: You need to remove the odata.context which will cause it to fail on another machine.

File: OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+            requestUri = string.Format("{0}/odata/Orders(Name='Order-2')", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string nameResponse = await response.Content.ReadAsStringAsync();
+
+            // query with another alternate key
+            requestUri = string.Format("{0}/odata/Orders(Token=75036B94-C836-4946-8CC8-054CF54060EC)", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string guidRespoonse = await response.Content.ReadAsStringAsync();
```
Comment (abkmr) [inline OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs]: Convert it to tokenResponse to have same naming convention as nameResponse. Or change both to tokenGuidResponse, nameStringResponse.

File: OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs (hunk 1 above)
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs]: Yes. Thanks.

File: OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs (hunk 3 above)
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs]: Yes. Thanks

**Comment by xuzhg** [comment]
@abkmr I have updated the PR. Thanks.

**Comment by xuzhg** [comment]
Merged. Thanks.

**Pull request closed without merging**